apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java

```diff
@@ -100,70 +99,56 @@ public GeneratorServiceImpl(PropertiesHolder propertiesHolder,
 
     @Override
     public GeneratorMemoryEntity startForging(byte[] keySeed) {
+        long start = System.currentTimeMillis();
         if (generators.size() >= MAX_FORGERS) {
             throw new RuntimeException("Cannot forge with more than " + MAX_FORGERS + " accounts on the same node");
         }
         byte[] publicKey = Crypto.getPublicKey(keySeed);
         long accountId = AccountService.getId(publicKey);
         GeneratorMemoryEntity generator = new GeneratorMemoryEntity(keySeed, publicKey, accountId);
-        globalSync.updateLock();
-        try {
-            if (blockchain.getHeight() >= blockchainConfig.getLastKnownBlock()) {
-                this.setLastBlock(blockchain.getLastBlock(), generator);
-                if (generateBlocksTask != null) {
-                    generateBlocksTask.resetSortedForgers();
-                }
-            }
-        } finally {
-            globalSync.updateUnlock();
+        if (blockchain.getHeight() >= blockchainConfig.getLastKnownBlock()) {
+            this.setLastBlock(blockchain.getLastBlock(), generator);
+            scheduleAsyncForgersReset();
         }
         GeneratorMemoryEntity old = generators.putIfAbsent(generator.getAccountId(), generator);
         if (old != null) {
-            log.debug(old + " is already forging");
+            log.debug("{} is already forging ({} ms)", old, (System.currentTimeMillis() - start));
             return old;
         }
-        log.debug(generator + " started");
+        log.debug("{} started ({} ms)", generator, (System.currentTimeMillis() - start));
         return generator;
     }
 
     @Override
     public GeneratorMemoryEntity stopForging(byte[] keySeed) {
-        long start = System.currentTimeMillis();
         GeneratorMemoryEntity generator = generators.remove(Convert.getId(Crypto.getPublicKey(keySeed)));
-        if (generator != null && generateBlocksTask != null) {
-            globalSync.updateLock();
-            try {
-                generateBlocksTask.resetSortedForgers();
-            } finally {
-                globalSync.updateUnlock();
-            }
-            log.debug(generator + " stopped ({} ms)", (System.currentTimeMillis() - start));
+        if (generator != null) {
+            log.debug(generator + ": Stop Forging command added to the queue.");
+            scheduleAsyncForgersReset();
```

Comment:
Probably parameter injection into a string pattern with '{} my log string' would be more consistent approach

Reply:
fixed